components/CheckDocumentation/vacuum/XminHorizon.tsx

```diff
@@ -0,0 +1,205 @@
+import { useCodeBlock } from "../../CodeBlock";
+
+const XminHorizonTrigger: React.FunctionComponent<CheckTriggerProps> = ({
+  config,
+}) => {
+  const threshold = config.settings["behind_days"] as number;
+  const dayPluralized = threshold === 1 ? "day" : "days";
+  return (
+    <p>
+      Detects when the xmin horizon on the server has not progressed for the
+      last <code>{threshold}</code> {dayPluralized} and creates an issue with
+      severity "info". The issue will be created even if no VACUUM is currently
+      blocked by this, as this will potentially block any future VACUUMs.
+      Resolves once the xmin horizon makes some progress.
+    </p>
+  );
+};
+
+const XminHorizonGuidance: React.FunctionComponent<CheckGuidanceProps> = ({
+  issue,
+}) => {
+  const CodeBlock = useCodeBlock();
+  const issueDetails = issue && JSON.parse(issue.detailsJson);
+  const heldBackBy = issueDetails && issueDetails["held_back_by"];
+  let byBackend = null;
+  let byReplicationSlot = null;
+  let byReplicationSlotCatalog = null;
+  let byStandby = null;
+  let byPreparedXact = null;
+  if (heldBackBy) {
+    byBackend = heldBackBy.find((v) => v["type"] === "backend");
+    byReplicationSlot = heldBackBy.find(
+      (v) => v["type"] === "replication_slot"
+    );
+    byReplicationSlotCatalog = heldBackBy.find(
+      (v) => v["type"] === "replication_slot_catalog"
+    );
+    byStandby = heldBackBy.find((v) => v["type"] === "standby");
+    byPreparedXact = heldBackBy.find((v) => v["type"] === "prepared_xact");
+  }
+  return (
+    <div>
+      <h4>Impact</h4>
+      <p>VACUUM is potentially blocked by the xmin horizon.</p>
+      <p>
+        The xmin horizon tells you until which point the vacuum process can
+        clean up dead rows. When this value is behind and not making progress,
+        it's very likely that VACUUM can't clean up dead rows, hence blocking
+        VACUUM.
+      </p>
+      <p>
+        When VACUUM is blocked and dead rows can't be cleaned, it can result to
+        the unnecessary table bloat or slow queries.
+      </p>
+      <h4>Common Causes</h4>
+      <ul>
+        <li>
+          <h5>Long-running transactions</h5>
+          <p>
+            When there is a long running transaction, you can't clean up rows
+            that the transaction can see, even if it already became a dead row.
+          </p>
+        </li>
+        <li>
+          <h5>Delayed/bad replication slots</h5>
+          <p>
+            When the replication is delayed or the standby server is down, the
+            oldest transaction that the replication slot needs the database to
+            retain can be "stuck" hence hold back of the xmin horizon can
+            happen. This can also happen to the system catalogs.
+          </p>
+        </li>
+        <li>
+          <h5>Long-running transactions on standbys</h5>
+          <p>
+            When <code>hot_standby_feedback</code> is on, queries on standbys
```

Comment:
We could use a SettingLink here, but that's probably not what we want because we'd be linking to the value of the setting on the primary, not on the replica, right? Just sanity-checking my own reasoning.

Reply:
Yeah you're right, linking to primary's `hot_standby_feedback` here is meaningless. However, I think you have a good point re: linking to some settings, or we actually should be able to eliminate some suggestions based on how the current server is like.

I'll look into more for this one, I've actually been confused around here and it would be great to get the complete understanding, as well as showing better/proper explanation here.

Reply:
We actually should be able to eliminate all unrelated suggestions, so I changed the structure.